demo/admin/src/common/Routes.tsx

```diff
@@ -0,0 +1,43 @@
+import { MasterLayout, RouteWithErrorBoundary } from "@comet/admin";
+import { SitePreview, useCurrentUser } from "@comet/cms-admin";
+import { CircularProgress } from "@mui/material";
+import ContentScopeProvider from "@src/common/ContentScopeProvider";
+import * as React from "react";
+import { Redirect, Route, RouteProps, Switch } from "react-router-dom";
+
+import MasterHeader from "./MasterHeader";
+import MasterMenu from "./MasterMenu";
+import { pages } from "./pages/pages";
+
+export const Routes: React.FC = () => {
+    const user = useCurrentUser();
+    if (!user) return <CircularProgress />;
+
+    const routesForUser: RouteProps[] = pages
+        .flatMap((page) => [page.route && { ...page.route, requiredPermission: page.requiredPermission }, ...(page.routes ?? [])])
+        .filter((route) => route && user.isAllowed(route.requiredPermission)) as RouteProps[];
```

Comment:
is it required to filter the routes at all?

Reply:
I don't see a reason against it. We have to filter the menuItems anyway so we can also filter the routes.

Reply:
Wouldn't a user be able to access the route by manually entering the URL if we don't filter them here?

Reply:
> Wouldn't a user be able to access the route by manually entering the URL if we don't filter them here?

yes, but that is very easy to bypass; after all the api needs to be secured correctly.

on the other hand it doesn't hurt and gives a bit more security :+1: